src/LanguageServer/Impl/Implementation/Server.Documents.cs

```diff
@@ -78,5 +82,10 @@ public sealed partial class Server {
             _log?.Log(TraceEventType.Error, $"Unable to find document {uri}");
             return null;
         }
+
+        private void DocNewAstEvent(object sender, EventArgs _) {
+            var document = (IDocument)sender;
+            _indexManager.ReIndexFile(document.Uri.AbsolutePath, document);
```

Comment:
This event comes on every keystroke. Will yield 100% CPU. Should be throttled to idle + some time, like 300ms+